docs/300_OBSERVER.md

```diff
@@ -0,0 +1,76 @@
+# Generic Observer
+An `Observer` evaluates upkeep state from a chain registry and collects upkeeps
+that need to be performed along with data required for comparison and reporting.
+
+Examples can include:
+- polling observer (polls the chain at each block)
+- event trigger observer (logs can trigger observations)
+- scheduled observer (polling occurs on a defined schedule like cron)
+
+## Observer Rules
+- an observer should only be used for observation triggers and timing
+- an observer should be unaware of registry specific types
+- an observer should be chain agnostic
+- an observer should interact with a registry through an interface
+- an observer can define its own dependency interfaces
+- an observer should use a `Coordinator` to track in-flight state
+- all observers used by the same plugin instance should use the same registry
```

Comment:
I think we can start with the assumption that there will be one observer for a plugin and generalise it later when we see a need

Reply:
For now, I think that is what we will do. Looping through an array of observers vs reading a single one won't be much different.